python/pyspark/sql.py

```diff
@@ -2136,9 +2136,9 @@ def __getitem__(self, item):
 
     def __getattr__(self, name):
         """ Return the column by given name """
-        if isinstance(name, basestring):
-            return Column(self._jdf.apply(name))
-        raise AttributeError
+        if name.startswith("__"):
```

Comment:
`Row` [already had this check](https://github.com/JoshRosen/spark/blob/SPARK-5464-python-dataframe-help-command/python/pyspark/sql.py#L1773) in its `__getattr__` call, so I think it's safe to do the same thing here, too.